pandas/io/json/_normalize.py

```diff
@@ -148,13 +149,18 @@ def _normalise_json(
     if isinstance(data, dict):
         for key, value in data.items():
             new_key = f"{key_string}{separator}{key}"
+
+            if not key_string:
+                if sys.version_info < (3, 9):
```

Comment:
@MarcoGorelli if we use `if not PY310` where `PY310` is from `pandas.compat` would pyupgrade still flag this?

Reply:
it wouldn't, no, pyupgrade just does static analysis (it wouldn't know what the symbol `PY310` means) - in fact, I was kinda tempted to replace all the `PY310` and other `pandas.compat` constants with `sys.version_info` checks, so we don't need to remember what to clean up when dropping versions each year

Reply:
I would be open to this change.